### ISSUE TITLE
REQUEST [error] Error (error) while connecting to http://127.0.0.1:34628/event_notification

### ISSUE THREAD

**Issue by zhuzhongshu** [abingham/emacs-ycmd#269]
Hello, 

The emacs-ycmd cannot work on my machine. When I run `global-ycmd-mode` to turn on the ycmd mode, it will display an erro message like "REQUEST [error] Error (error) while connecting to http://127.0.0.1:34628/event_notification"  in the mini buffer. And this error message  appears whenever I try to invoke the ycmd server. For example, when I type `std::` .

**Comment by abingham** [comment]
Please post the contents of your `*ycmd-server*` buffer. This will help us see if the server is functioning properly.

**Comment by zhuzhongshu** [comment]
Here is the contents of the ` ycmd-server`

2016-02-19 19:48:48,210 - DEBUG - Global extra conf not loaded or no function YcmCorePreload
serving on http://127.0.0.1:42989
2016-02-19 19:48:48,400 - INFO - Received event notification
2016-02-19 19:48:48,400 - DEBUG - Event name: FileReadyToParse
2016-02-19 19:48:48,400 - INFO - Adding buffer identifiers for file: /home/zzs/string_to_double.cpp
<h1>Critical error while processing request: /event_notification</h1>2016-02-19 19:48:48,927 - ERROR - Exception when serving /event_notification
Traceback (most recent call last):
  File "/home/zzs/.emacs.d/ycmd/third_party/waitress/waitress/channel.py", line 337, in service
    task.service()
  File "/home/zzs/.emacs.d/ycmd/third_party/waitress/waitress/task.py", line 173, in service
    self.execute()
  File "/home/zzs/.emacs.d/ycmd/third_party/waitress/waitress/task.py", line 392, in execute
    app_iter = self.channel.server.application(env, start_response)
  File "/home/zzs/.emacs.d/ycmd/third_party/bottle/bottle.py", line 977, in __call__
    return self.wsgi(environ, start_response)
  File "/home/zzs/.emacs.d/ycmd/third_party/bottle/bottle.py", line 972, in wsgi
    start_response('500 INTERNAL SERVER ERROR', headers, sys.exc_info())
  File "/home/zzs/.emacs.d/ycmd/third_party/waitress/waitress/task.py", line 354, in start_response
    reraise(exc_info[0], exc_info[1], exc_info[2])
  File "/home/zzs/.emacs.d/ycmd/third_party/bottle/bottle.py", line 958, in wsgi
    start_response(response._status_line, response.headerlist)
  File "/home/zzs/.emacs.d/ycmd/third_party/waitress/waitress/task.py", line 372, in start_response
    'Header name %r is not a string in %r' % (k, (k, v))
AssertionError: Header name u'X-Ycm-Hmac' is not a string in (u'X-Ycm-Hmac', 'iYr36d6sfV07Py/jjFmcUoWIVri4IrizT0dUb+cfCbc=')
2016-02-19 19:49:18,232 - INFO - Received health request
<h1>Critical error while processing request: /healthy</h1>2016-02-19 19:49:18,232 - ERROR - Exception when serving /healthy
Traceback (most recent call last):
  File "/home/zzs/.emacs.d/ycmd/third_party/waitress/waitress/channel.py", line 337, in service
    task.service()
  File "/home/zzs/.emacs.d/ycmd/third_party/waitress/waitress/task.py", line 173, in service
    self.execute()
  File "/home/zzs/.emacs.d/ycmd/third_party/waitress/waitress/task.py", line 392, in execute
    app_iter = self.channel.server.application(env, start_response)
  File "/home/zzs/.emacs.d/ycmd/third_party/bottle/bottle.py", line 977, in __call__
    return self.wsgi(environ, start_response)
  File "/home/zzs/.emacs.d/ycmd/third_party/bottle/bottle.py", line 972, in wsgi
    start_response('500 INTERNAL SERVER ERROR', headers, sys.exc_info())
  File "/home/zzs/.emacs.d/ycmd/third_party/waitress/waitress/task.py", line 354, in start_response
    reraise(exc_info[0], exc_info[1], exc_info[2])
  File "/home/zzs/.emacs.d/ycmd/third_party/bottle/bottle.py", line 958, in wsgi
    start_response(response._status_line, response.headerlist)
  File "/home/zzs/.emacs.d/ycmd/third_party/waitress/waitress/task.py", line 372, in start_response
    'Header name %r is not a string in %r' % (k, (k, v))
AssertionError: Header name u'X-Ycm-Hmac' is not a string in (u'X-Ycm-Hmac', 'CxLXL5UodXeGdSu2fynwyQ6MjFu2VCoQGIai4shkQnQ=')
2016-02-19 19:49:48,231 - INFO - Received health request
<h1>Critical error while processing request: /healthy</h1>2016-02-19 19:49:48,232 - ERROR - Exception when serving /healthy
Traceback (most recent call last):
  File "/home/zzs/.emacs.d/ycmd/third_party/waitress/waitress/channel.py", line 337, in service
    task.service()
  File "/home/zzs/.emacs.d/ycmd/third_party/waitress/waitress/task.py", line 173, in service
    self.execute()
  File "/home/zzs/.emacs.d/ycmd/third_party/waitress/waitress/task.py", line 392, in execute
    app_iter = self.channel.server.application(env, start_response)
  File "/home/zzs/.emacs.d/ycmd/third_party/bottle/bottle.py", line 977, in __call__
    return self.wsgi(environ, start_response)
  File "/home/zzs/.emacs.d/ycmd/third_party/bottle/bottle.py", line 972, in wsgi
    start_response('500 INTERNAL SERVER ERROR', headers, sys.exc_info())
  File "/home/zzs/.emacs.d/ycmd/third_party/waitress/waitress/task.py", line 354, in start_response
    reraise(exc_info[0], exc_info[1], exc_info[2])
  File "/home/zzs/.emacs.d/ycmd/third_party/bottle/bottle.py", line 958, in wsgi
    start_response(response._status_line, response.headerlist)
  File "/home/zzs/.emacs.d/ycmd/third_party/waitress/waitress/task.py", line 372, in start_response
    'Header name %r is not a string in %r' % (k, (k, v))
AssertionError: Header name u'X-Ycm-Hmac' is not a string in (u'X-Ycm-Hmac', 'CxLXL5UodXeGdSu2fynwyQ6MjFu2VCoQGIai4shkQnQ=')
2016-02-19 19:50:18,231 - INFO - Received health request
<h1>Critical error while processing request: /healthy</h1>2016-02-19 19:50:18,231 - ERROR - Exception when serving /healthy
Traceback (most recent call last):
  File "/home/zzs/.emacs.d/ycmd/third_party/waitress/waitress/channel.py", line 337, in service
    task.service()
  File "/home/zzs/.emacs.d/ycmd/third_party/waitress/waitress/task.py", line 173, in service
    self.execute()
  File "/home/zzs/.emacs.d/ycmd/third_party/waitress/waitress/task.py", line 392, in execute
    app_iter = self.channel.server.application(env, start_response)
  File "/home/zzs/.emacs.d/ycmd/third_party/bottle/bottle.py", line 977, in __call__
    return self.wsgi(environ, start_response)
  File "/home/zzs/.emacs.d/ycmd/third_party/bottle/bottle.py", line 972, in wsgi
    start_response('500 INTERNAL SERVER ERROR', headers, sys.exc_info())
  File "/home/zzs/.emacs.d/ycmd/third_party/waitress/waitress/task.py", line 354, in start_response
    reraise(exc_info[0], exc_info[1], exc_info[2])
  File "/home/zzs/.emacs.d/ycmd/third_party/bottle/bottle.py", line 958, in wsgi
    start_response(response._status_line, response.headerlist)
  File "/home/zzs/.emacs.d/ycmd/third_party/waitress/waitress/task.py", line 372, in start_response
    'Header name %r is not a string in %r' % (k, (k, v))
AssertionError: Header name u'X-Ycm-Hmac' is not a string in (u'X-Ycm-Hmac', 'CxLXL5UodXeGdSu2fynwyQ6MjFu2VCoQGIai4shkQnQ=')

**Comment by abingham** [comment]
It looks like you're seeing [this problem](https://github.com/Valloric/ycmd/issues/377).

TL;DR - ycmd recently updated to support Python 3, and there are still some problems with Python 2.
